src/js/media-chrome-menu-button.js

```diff
@@ -0,0 +1,147 @@
+import './media-chrome-button.js';
+import './media-chrome-listbox.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { Window as window, Document as document } from './utils/server-safe-globals.js';
+import { MediaStateReceiverAttributes } from './constants.js';
+
+const template = document.createElement('template');
+template.innerHTML = `
+  <style>
+  :host {
+    display: block;
+  }
+
+  .hidden {
+    display: none;
+  }
+
+  media-chrome-button:not(:focus-visible) {
+    outline: none;
+  }
+  </style>
+
+  <media-chrome-button aria-haspopup="listbox" aria-controls="listbox">
+    <slot name="button-content"></slot>
+  </media-chrome-button>
+  <slot name="listbox" class="hidden">
+    <media-chrome-listbox id="listbox">
+      <slot></slot>
+    </media-chrome-listbox>
+  </slot>
+`;
+
+class MediaChromeMenuButton extends window.HTMLElement {
+  #handleClick;
+  #button;
+  #listbox;
+  #listboxSlot;
+  #expanded = false;
+
+  static get observedAttributes() {
+    return [
+      'disabled', MediaStateReceiverAttributes.MEDIA_CONTROLLER,
+    ];
+  }
+
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const buttonHTML = template.content.cloneNode(true);
+    this.nativeEl = buttonHTML;
+
+    shadow.appendChild(buttonHTML);
+
+    this.#handleClick = this.#handleClick_.bind(this);
+
+    this.#button = this.shadowRoot.querySelector('media-chrome-button');
+    this.#listbox = this.shadowRoot.querySelector('media-chrome-listbox');
+
+    this.#listboxSlot = this.shadowRoot.querySelector('slot[name=listbox]');
+    this.#listboxSlot.addEventListener('slotchange', () => {
+      const assignedListbox = this.#listboxSlot.assignedElements()[0];
+
+      if (assignedListbox) {
+        this.#listbox = this.#listboxSlot.assignedElements()[0];
+      }
+    });
+  }
+
+  #handleClick_() {
+    this.#listboxSlot.classList.toggle('hidden');
+    this.#expanded = !this.#expanded;
```

Comment:
suggestion(non-blocking): Consider moving `this.#expanded = !this.#expanded` into `#toggleExpanded()`